cmd/cosmos-sdk-cli/cmd/init.go

```diff
@@ -0,0 +1,128 @@
+package cmd
+
+import (
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/spf13/cobra"
+	"path/filepath"
+)
+
+func init() {
+	initCmd.Flags().StringVarP(&remoteProjectPath, "project-path", "p", "", "Remote project path. eg: github.com/your_user_name/project_name")
+	rootCmd.AddCommand(initCmd)
+}
+
+func resolveProjectPath(remoteProjectPath string) string {
+	gopath := os.Getenv("GOPATH")
+	if gopath == "" {
+		gopath = build.Default.GOPATH
+		// Use $HOME/go
+	}
+	return gopath + string(os.PathSeparator) + "src" + string(os.PathSeparator) + remoteProjectPath
+}
+
+var remoteBasecoinPath = "github.com/cosmos/cosmos-sdk/examples/basecoin"
+
+func copyBasecoinTemplate(projectName string, projectPath string, remoteProjectPath string) {
+	basecoinProjectPath := resolveProjectPath(remoteBasecoinPath)
+	filepath.Walk(basecoinProjectPath, func(path string, f os.FileInfo, err error) error {
+		if !f.IsDir() {
+			data, err := ioutil.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			contents := string(data)
+			// Extract relative file path eg: app/app.go instead of /Users/..../github.com/cosmos/...examples/basecoin/app/app.go
+			relativeFilePath := path[len(basecoinProjectPath)+1:]
+			// Evaluating the filepath in the new project folder
+			projectFilePath := projectPath + string(os.PathSeparator) + relativeFilePath
+			lengthOfRootDir := strings.LastIndex(projectFilePath, string(os.PathSeparator))
+			// Extracting the path of root directory from the filepath
+			rootDir := projectFilePath[0:lengthOfRootDir]
+			// Creating the required directory first
+			os.MkdirAll(rootDir, os.ModePerm)
+			fmt.Println("Creating " + projectFilePath)
+			// Writing the contents to a file in the project folder
+			ioutil.WriteFile(projectFilePath, []byte(contents), os.ModePerm)
+		}
+		return nil
+	})
+}
+
+func createGopkg(projectPath string) {
+	// Create gopkg.toml file
+	dependencies := map[string]string{
+		"github.com/cosmos/cosmos-sdk": version.Version,
+	}
+	overrides := map[string]string{
+		"github.com/golang/protobuf": "1.1.0",
+	}
+	contents := ""
+	for dependency, version := range dependencies {
+		contents += "[[constraint]]\n\tname = \"" + dependency + "\"\n\tversion = \"" + version + "\"\n\n"
+	}
+	for dependency, version := range overrides {
+		contents += "[[override]]\n\tname = \"" + dependency + "\"\n\tversion = \"=" + version + "\"\n\n"
+	}
+	contents += "[prune]\n\tgo-tests = true\n\tunused-packages = true"
+	ioutil.WriteFile(projectPath+"/Gopkg.toml", []byte(contents), os.ModePerm)
+}
+
+func createMakefile(projectPath string) {
```

Comment:
Hrmm, we're deviating from our current tools structure for linting. I'm conflicted on whether or not we should auto include our same linting structure. 

We can definitely postpone a decision on this, perhaps make a comment that we should add linters and the `/tools/` directory in the future. (Unless you'd like to that in this PR)

Reply:
For the start, I would not add it. I'll just add a comment on it as you suggested.